torch/_higher_order_ops/auto_functionalize.py

```diff
@@ -234,23 +417,81 @@ def do_auto_functionalize(
             normalized_kwargs[arg.name] = arg.default_value
 
     unwrapped_kwargs = ctx.unwrap_tensors(normalized_kwargs)  # type: ignore[arg-type]
+
     if "self" in unwrapped_kwargs or "self_" in unwrapped_kwargs:
         warnings.warn(
             "Using `self` or `self_` as an argument in the definition of custom ops may lead to ambiguous parsing. "
             "Please consider using a different name for this argument to avoid potential issues."
         )
+    # List of the name of args that get mutated (according to the schema)
+    mutable_args_names, mutable_args_types = get_mutable_args(op)
+
+    # A list of all bases of mutable args without duplication
+    all_basis = []
+    all_basis_addresses: list[int] = []
+
+    # Map arg_name to the index of its base in all_basis.
+    arg_to_base_index: Dict[str, Any] = {}
+    for arg_name in mutable_args_names:
+        arg = normalized_kwargs[arg_name]
+        if arg is None:
+            continue
+
+        if isinstance(arg, list):
+            arg_to_base_index[arg_name] = {}
+            for i, tensor in enumerate(arg):
+                if tensor is None:
+                    arg_to_base_index[arg_name].append(None)
+                    continue
+
+                base = tensor if tensor._base is None else tensor._base
+
+                if not all_basis_addresses.__contains__(base._cdata):
+                    all_basis_addresses.append(base._cdata)
+                    all_basis.append(base)
+                    arg_to_base_index[arg_name][i] = len(all_basis) - 1
+                else:
+                    arg_to_base_index[arg_name][i] = all_basis_addresses.index(
+                        base._cdata
+                    )
+
+        else:
+            base = arg if arg._base is None else arg._base
```

Comment:
Can you run one of the tests in test/dynamo/test_misc.py under torch.inference_mode ? I'm not sure if arg._base is available in inference_mode, if it is not then we need to figure out some way to get it (Functionalization definitely has a way to do this if it has this problem).

Reply:
i think i did and it worked will add test